code/game/objects/structures/door_assembly.dm

```diff
@@ -258,6 +258,27 @@
 		deconstruct(TRUE)
 	update_icon(UPDATE_OVERLAYS)
 
+/obj/structure/door_assembly/multitool_act(mob/user, obj/item/I)
+	if(noglass)
+		return
+	. = TRUE
+	if(state != AIRLOCK_ASSEMBLY_NEEDS_SCREWDRIVER)
+		to_chat(user, "<span class='notice'>The airlock assembly needs its electronics before you can configure the electrochromic windows.</span>")
+		return
+	if(!glass && !noglass)
+		to_chat(user, "<span class='notice'>The airlock assembly needs glass before you can configure the electrochromic windows.</span>")
+		return
+
+	if(!I.tool_use_check(user, 0))
+		return
+	user.visible_message("[user] is configuring the windows in the airlock assembly...", "You start to configure the windows in the airlock assembly...")
+	if(!I.use_tool(src, user, 40, volume = I.tool_volume) || state != AIRLOCK_ASSEMBLY_NEEDS_SCREWDRIVER)
```

Comment:
```suggestion
	if(!I.use_tool(src, user, 4 SECONDS, volume = I.tool_volume) || state != AIRLOCK_ASSEMBLY_NEEDS_SCREWDRIVER)
```

You might also want to move the second check before the use_tool so it doesn't have to wait for the do_after.

Reply:
Doesn't that mean after it is done with the tool progress, it'll just ignore the current state of the assembly? That'd be different from the other tool acts.

Reply:
Sorry, I didn't want to get too busy with the suggestions. I just meant something like

```
if(state != AIRLOCK_ASSEMBLY_NEEDS_SCREWDRIVER)
    return

if(!I.use_tool(src, user, 4 SECONDS, volume = I.tool_volume))
    return

Reply:
Fiddling around, I realized it should have checked for electronics instead. Also, it does need to be part of the same check. If I put it before or after the use tool, it'd just ignore it altogether.